shared/macros/10-warning.jinja

```diff
@@ -44,3 +44,24 @@
 - general: |-
     There is no remediation for this besides re-compiling the kernel with the appropriate value for the config.
 {{% endmacro %}}
+
+{{#
+    Includes a deprecation warning in rules which are no longer recommended because of some
+    changes in standard requirements or similar reasons.
+
+:param reason: General text explaining the reason of deprecation, i.e. standard changes etc
+:type reason: str
+:param release: Release number since the rule is officially deprecated.
+:type release: str
+
+#}}
+{{% macro warning_rule_deprecated_cause(reason, release='') %}}
```

Comment:
This macro is a very similar copy of the `warning_rule_deprecated_by`.  Why not using the existing macro? If there is no rule to replace the `deprecated` rule maybe it would be easier to extend the existing macro so we keep the deprecation process more consistent.

Reply:
Good idea will do